File: race-schedule-service/src/main/java/org/example/racescheduleservice/service/SeasonService.java
package org.example.racescheduleservice.service;

import org.example.racescheduleservice.model.Season;
import org.example.racescheduleservice.repository.SeasonRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class SeasonService {

    @Autowired
    private SeasonRepository seasonRepository;

    // Lấy tất cả mùa giải
    public List<Season> getAllSeasons() {
        return seasonRepository.findAll();
    }

    // Lấy mùa giải theo id
    public Optional<Season> getSeasonById(String id) {
        return seasonRepository.findById(id);
    }
}

File: f1-management-frontend/src/services/seasonService.js
import api from './api';
// Lấy tất cả mùa giải
export const getAllSeasons = async () => {
  try {
    const response = await api.get('/seasons');
    return response.data;
  } catch (error) {
    console.error('Error fetching seasons:', error);
    throw error;
  }
};
// Lấy mùa giải theo id
export const getSeasonById = async (id) => {
  try {
    const response = await api.get(`/seasons/${id}`);
    return response.data;
  } catch (error) {
    console.error(`Error fetching season with id ${id}:`, error);
    throw error;
  }
};
